_posts/2018-12-20-so-long-and-thanks.md

```diff
@@ -0,0 +1,131 @@
+Furthermore, I would have asked so many questions about how the apps work from a business
+perspective by shadowing the stakeholders (underwriters, prequal analysts) and seeing the overall workflow
+and what issues there were with it. I only got a good feel for ONE part of the underwriting app
+after struggling on a card (described in detail later) and asking lots of questions directly to an
+underwriter. As weeks passed by and I got more comfortable at the job, 
+I got better about leaving aside that feeling and asking the question anyways - 
+but I do wish that I had done it sooner.
+
+If I had to do it all over again, I would've asked about five times more questions than I did
+regardless of how basic or obvious the answers seemed.
+Funny thing is, I've heard this advice over and over again
+and yet it fell on deaf ears for me until I went through the experience myself. But I know now
+for the future that I might end up a bit annoying sometimes but at least I'll learn a lot!
+
+# Always in Flux
+A rather unique experience I had during my time at Forward Financing was quickly adjusting
+to the constant change. From when I joined in July to a couple months later in
+December, the company went through a few big changes. We had (and still have) a shortage
+of product managers which taught me how important they are in keeping the
+software development lifecycle (SDLC) running smoothly especially with scoping out tasks and minimizing
+the friction between stakeholders and engineers. 
+
+I would like to start of by talking about a rather significant change in our engineering process where our
+sprint of three weeks + a few days reserved for what we call slack time - a short period where the team
+works on tech debt and projects that are not of immediate business concern -
+was changed to two weeks with no slack time. However, I feel that this did not pan
+out too well due to these reasons:
+* Meeting durations were not modified to reflect the reduction in sprint duration
+which meant that on the days of those meetings, productivity was hindered due to those
+meetings acting as distractions.
+* In our workflow, a lot of the tasks require collaboration from multiple teams.
+Sometimes the back-and-forth between coworkers working on a feature together takes time 
+and with unavoidable external factors (eg. QA team backlogged, Heroku being down),
+engineers could lose a day or two of productive work. Therefore, the impact of a lost day
+in a two-week sprint compared to a three-week sprint proves to be much more significant.
+* Due to the prior points and the lack of slack time to work on tech debt, the non-immediate
+yet non-trivial tech debt tasks were piling up. Ideally, there was to be a portion of
+time during the sprint dedicated towards tackling technical debt; however, this did not work out
+due to the above reducing the time allotted for prioritized feature work, leaving no room for it.
```

Comment:
Kind of getting into the weeds here with our own internal housekeeping. I don't think you really need to get too specific here, but you can be more general in saying that fine tuning our sprint mechanics was a challenge.